### PR TITLE
feat(Aside): add option to hide hanging rule from Aside component

### LINKED ISSUE #457: [Aside] Support no-rule option
The Aside component (https://gatsby-theme-carbon.now.sh/components/Aside/) comes with a hanging rule. Design would like an option to not have the rule. 

image

### PR DESCRIPTION
**Pull request by luizamendes**
Closes #457

Add support to no rule option

#### Changelog

**New**

- Now the Aside component can receive a `hideRule` boolean prop that hides the hanging rule

### CODE REVIEW

**Comment by jnm2377** [comment]
Hi! Thanks for opening this PR. Code-wise, it looks good to me. We just need to get design sign off on this. :)

**Comment by luizamendes** [comment]
@jnm2377 No problem, thank you for the feedback. If you could consider adding the `"hacktoberfest-accepted"` label so that it count towards the event, I would appreciate. Thanks.

File: packages/gatsby-theme-carbon/src/components/Aside/Aside.module.scss [hunk 1]
```diff
@@ -24,6 +24,10 @@
   background: $ui-05;
 }
 
+.asideNoRule:before {
```
Comment (vpicone) [inline packages/gatsby-theme-carbon/src/components/Aside/Aside.module.scss:27]:
```suggestion
.asideNoRule::before {
```

**Review by vpicone — APPROVED** [review]
Looks great!

**Review by vpicone — APPROVED** [review]
Looks good to me, do you mind adding an example using the prop to the MDX?

**Comment by luizamendes** [comment]
@vpicone Just added an example, following the examples from other pages.

**Review by jeanservaas — APPROVED** [review]

**Review by jnm2377 — APPROVED** [review]

**Pull request merged**